library/psa_crypto.c

```diff
@@ -1923,7 +1923,7 @@ psa_status_t psa_copy_key( mbedtls_svc_key_id_t source_key,
     defined(MBEDTLS_PSA_BUILTIN_ALG_RSA_OAEP) || \
     defined(MBEDTLS_PSA_BUILTIN_ALG_RSA_PSS) || \
     defined(MBEDTLS_PSA_BUILTIN_ALG_DETERMINISTIC_ECDSA)
-static const mbedtls_md_info_t *mbedtls_md_info_from_psa( psa_algorithm_t alg )
+const mbedtls_md_info_t *mbedtls_md_info_from_psa( psa_algorithm_t alg )
```

Comment:
I noticed all three invocations of `mbedtls_md_info_from_psa` that used to be in `psa_crypto.c` don't actually want the full information, but only the `mbedtls_md_t alg_type`. Wouldn't it make sense to rework this to not be dependent on actual 'builtin' algorithm presence, but rather have an always-defined mapping from `psa_algorithm_t` to `mbedtls_md_t`?

Reply:
FYI, I'm currently reworking the hashing dispatch. PR soon (tm)

Reply:
> I noticed all three invocations of `mbedtls_md_info_from_psa` that used to be in `psa_crypto.c` don't actually want the full information, but only the `mbedtls_md_t alg_type`. Wouldn't it make sense to rework this to not be dependent on actual 'builtin' algorithm presence, but rather have an always-defined mapping from `psa_algorithm_t` to `mbedtls_md_t`?

Like this, it makes sense to me. There are maybe hash features not implemented yet for which more than just the type would be needed though.